### DISCUSSION TITLE
Can ESP Now be enabled/setup on grblHAL ESP32 driver?

### DISCUSSION THREAD

**Discussion by gwcSoftware** [grblHAL/ESP32#126] [General]
Would like to communicate with another ESP32 device as a job is running.

Can ESP NOW be enabled/setup to send updates to another ESP32 device?

I have the Openbuilds BlackboxX32 that uses the ESP32-WROOM-32.

If so, any info on how to do this would be appreciated.

**Comment by terjeio** [comment]
Lots of info [here](https://forum.v1e.com/t/wireless-pendant-using-esp-now/35999).

**Reply by gwcSoftware** [reply]
Thanks, I'll have a look.

**Reply by gwcSoftware** [reply]
I checked out the link. That was quite a project but I didn't see any detailed info on how he added to grblHAL ESP32.

**Comment by gwcSoftware** [comment]
Well, I couldn't give up! I finally have been able to build ESP32 with my new code and ESP-NOW gets initialized and communicates with my other ESP device. 

However, when I try to connect Openbuilds Control to my ESP32 device with the new build, Openbuilds Control doesn't see grblHAL and closes the COM port. It looks like my new code is blocking this connection. I have made sure that the COM port is not in use and have tried connecting the device on another laptop also. Same issue, Openbuilds does not see grbl running anywhere.

**Reply by gwcSoftware** [reply]
Found my problem! Being new to this, I had created a separate ESP-IDF App for testing and when I copied the .c/.h files over into grblHAL ESP32, I forgot to rename the  app_main() function. 

After renaming and fixing some of the config variables specifying my files, I was able to build and install to the ESP device and connect using Openbuilds Control without problems.
